lib/order.rb

```diff
@@ -1,18 +1,57 @@
-module Grocery
-  class Order
-    attr_reader :id, :products
+require 'csv'
 
-    def initialize(id, products)
-      @id = id
-      @products = products
-    end
+module Grocery	
+	class Order
+		attr_reader :id, :products  
+		
+		#ID - string; products - map
+		def initialize(id, products)
+		  @id = id
+		  @products = products
+		end
 
-    def total
-      # TODO: implement total
-    end
+		def self.all()
+			myOrders = []
+			#row[0] = id | row[1] = orders
+			CSV.foreach('../support/orders.csv') { |row|
+				order_prodcuts = {}
+				order_id = row[0] #this is the ID
+				row[1].split(';').each { |i| #this splits the order string into individual name/cost pairs
+					single_item = i.split(':')#separates the name from the cost
+					cost = single_item[1]
+					name = single_item[0]
```

Comment:
I like the logic you've written to parse the list of products. Your local variables (`single_item`, `cost` and `name`) are clearly named and make it easy to tell what's going on.